test/qr_test.go

```diff
@@ -11,7 +11,8 @@ import (
 )
 
 func TestQRProof(t *testing.T) {
-	log.TurnOff() // otherwise too many communication messages
+	prevLogger := client.Logger
+	client.Logger = log.NewNullLogger()
```

Comment:
Now I see why you're exporting the logger. I was expecting a solution with implementing `func setLogger(newLogger)` on the client. I realize you probably did it this way since I was annoyed when seeing logger-related functions in client/client.go, but `setLogger` is different IMO. Anyway, do as you please, you can leave it with like this.

Reply:
Well yeah, this exactly, you basically answered yourself :smile: 
I have to admit I wasn't too fond of exporting the logger, so I'll implement your suggestion.